### PR TITLE
Update to torch1.6

### LINKED ISSUE #555: PyTorch 1.6 function name modification
Error：
```
/home/unix/miniconda3/lib/python3.8/site-packages/torch/cuda/memory.py:344: FutureWarning: torch.cuda.memory_cached has been renamed to torch.cuda.memory_reserved
```

raw:
```
# for https://github.com/ultralytics/yolov5/blob/adaf1d8ac109f5c0977c27ceb3c2165a9c619cf2/train.py#L315
                mem = '%.3gG' % (torch.cuda.memory_cached() / 1E9 if torch.cuda.is_available() else 0)  # (GB)
```

new:
```
                mem = '%.3gG' % (torch.cuda.memory_reserved() / 1E9 if torch.cuda.is_available() else 0)  # (GB)
```

### PR DESCRIPTION
**Pull request by Lornatang**
PyTorch have `Automatic Mixed Precision (AMP) Training`.

fix #555 and #557 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Upgraded mixed precision training support with native PyTorch AMP (Automatic Mixed Precision) tools and refined model configuration files.

### 📊 Key Changes
- 🚀 Switched to native PyTorch AMP by importing `torch.cuda.amp` and using its `autocast` feature for mixed precision.
- 🛠️ Updated model configuration files for YOLOv3-SPP, YOLOv5 FPN, and YOLOv5 PANet, reducing complexity and redundant layers.
- 🧽 Removed Apex dependency from the requirements, as native AMP is now used.
- ✨ `GradScaler` now implemented for dynamic scaling of loss, replacing Apex's scaler for better compatibility and support.

### 🎯 Purpose & Impact
- 🔍 Improves YOLO models by streamlining the configurations for better maintainability and potentially improved performance.
- 📈 Enhances training speed and reduces memory usage by leveraging native PyTorch capabilities for mixed precision training, making the process more efficient.
- 🔨 Facilitates easier setup and lower prerequisites for new users, as the complex Apex installation is no longer necessary.

### CODE REVIEW

**Comment by Lornatang** [comment]
@glenn-jocher I found a conflict problem, you can ignore my previous two submissions. This time the main PR focuses on fixing the problem of PyTorch 1.6 version, not the model configuration file problem. sorry.

**Comment by glenn-jocher** [comment]
@Lornatang thanks buddy. It looks like you beat me to it. One thing you have to do though is to rebase your branch to the current master. This will remove the changes to the yaml's etc and bring your branch up to speed with master. I put these recommendations in the pr greetings action, but it never seems to run so I'll just paste them here:

- Verify your PR is up-to-date with origin/master. If your PR is behind origin/master update by running the following, replacing 'feature' with the name of your local branch:

```bash
git remote add upstream https://github.com/ultralytics/yolov5.git
git fetch upstream
git checkout feature  # <----- replace 'feature' with local branch name
git rebase upstream/master
git push -u origin -f
```
